search-app/src/cmr/search/data/complex_to_simple_converters/has_granules_or_cwic.clj

```diff
@@ -18,9 +18,15 @@
     (let [has-granules-or-cwic-map (has-granules-or-cwic-base/get-has-granules-or-cwic-map context)
           has-granules-map (has-granules-base/get-has-granules-map context)
           concept-ids (map key (filter val has-granules-map))
-          condition (cqm/string-conditions :concept-id concept-ids true)
+          condition (if (seq concept-ids)
+                      (cqm/string-conditions :concept-id concept-ids true)
+                      ;; when concept-ids are empty, string-conditions throw internal errors.
+                      ;; replace it with a non-existing concept-id to make sure nothing is returned using the condition..
+                      (cqm/string-conditions :concept-id ["non-existing"] true))
           or-cwic-concept-ids (map key (filter val has-granules-or-cwic-map))
-          or-cwic-condition (cqm/string-conditions :concept-id or-cwic-concept-ids true)]
+          or-cwic-condition (if (seq or-cwic-concept-ids)
+                              (cqm/string-conditions :concept-id or-cwic-concept-ids true)
+                              (cqm/string-conditions :concept-id ["non-existing"] true))]
```

Comment:
Why not use `when` if the second condition will always return true? If there are no values, we shouldn't add the condition to query just to force something

Reply:
when won't work, but match-none will.  Thanks.